app/src/app/api/v0/city/invite/[invite]/route.ts

```diff
@@ -5,61 +5,47 @@ import { Session } from "next-auth";
 import { NextRequest, NextResponse } from "next/server";
 import jwt from "jsonwebtoken";
 
-export const GET = apiHandler(
-  async (
-    req: NextRequest,
-    context: { session?: Session; params: Record<string, string> },
-  ) => {
-    const { params, session } = context;
+export const GET = apiHandler(async (_req, { params, session }) => {
+  const invite = await db.models.CityInvite.findOne({
+    where: {
+      id: params.invite,
+    },
+  });
 
-    if (!session) throw new createHttpError.Unauthorized("Unauthorized");
+  if (!invite) {
+    throw new createHttpError.NotFound("Not found");
+  }
 
-    const invite = await db.models.CityInvite.findOne({
-      where: {
-        id: params.invite,
-      },
-    });
+  const token = _req.nextUrl.searchParams.get("token");
+  const email = _req.nextUrl.searchParams.get("email");
```

Comment:
Email needs to be part of the token, otherwise any user obtaining the token could use the invite. If it's part of the signed JWT body, you can be sure that it's only valid for a single user. Could be their email or their user ID.

Reply:
You can still accept it as the search parameter like here, just make sure you verify it.

Reply:
I'm seeing you pass it into the token, but you don't compare the contents of the decrypted token to the passed email address.